Add event/timer functions

csssuf opened the pull request:
Add the `create_event` and `set_timer` functions to boot services, and add a `read_key_timeout` function to `Console` utilizing them.

orumin committed:
console: fix build error

lucab committed:
travis: add stable/beta/nightly matrix

lucab committed:
uefi: remove unused imports

csssuf committed:
bootservices: Add create_timer and set_timer

csssuf committed:
bootservices: instantiate Event correctly

bgilbert commented on src/bootservices.rs:
```diff
@@ -71,6 +73,23 @@ impl BootServices {
+        Ok(event)
+    }
+
+    pub fn set_timer(&self, event: Event, delay_type: TimerDelay, delay_ms: u64) -> Status {
```
By comparison, the `stall()` wrapper takes microseconds.  I'd suggest either using the underlying units or naming this `set_timer_ms`.

bgilbert commented on src/console.rs:
```diff
@@ -200,5 +205,31 @@ impl SimpleTextInput for Console {
         }
     }
+
+    fn read_key_timeout(&self, timeout_millis: u64) -> Option<Result<InputKey, Status>> {
```
I'm not sure this belongs in library code.  The set of events that are interesting to an application at a particular moment is typically specific to the application.  The function won't work for you, either, after you add serial support.

csssuf committed:
bootservices: use 100ns increments in set_timer

This corresponds with the unit actually accepted by the UEFI set_timer
function.

csssuf committed:
console: expose input's wait_for_key event
